SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -275,6 +276,39 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 	}
 }
 
+void USpatialNetDriver::CreateServerWorkerEntity()
+{
+	const FString WorkerId = FString(TEXT("workerId:")) + Connection->GetWorkerId();
+	const WorkerRequirementSet WorkerIdPermission{ { WorkerId } };
+
+	WriteAclMap ComponentWriteAcl;
+	ComponentWriteAcl.Add(SpatialConstants::POSITION_COMPONENT_ID, WorkerIdPermission);
+	ComponentWriteAcl.Add(SpatialConstants::METADATA_COMPONENT_ID, WorkerIdPermission);
+	ComponentWriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, WorkerIdPermission);
+	ComponentWriteAcl.Add(SpatialConstants::INTEREST_COMPONENT_ID, WorkerIdPermission);
+
+	QueryConstraint constraint;
+	constraint.EntityIdConstraint = SpatialConstants::INITIAL_GLOBAL_STATE_MANAGER_ENTITY_ID;
+
+	Query query;
+	query.Constraint = constraint;
+	query.FullSnapshotResult = true;
+
+	ComponentInterest queries;
+	queries.Queries.Add(query);
+
+	Interest interest;
+	interest.ComponentInterestMap.Add(SpatialConstants::POSITION_COMPONENT_ID, queries);
+
+	TArray<Worker_ComponentData> Components;
+	Components.Add(Position().CreatePositionData());
+	Components.Add(Metadata(WorkerId).CreateMetadataData());
+	Components.Add(EntityAcl(WorkerIdPermission, ComponentWriteAcl).CreateEntityAclData());
+	Components.Add(interest.CreateInterestData());
+
+	Connection->SendCreateEntityRequest(MoveTemp(Components), nullptr);
```

Comment:
What to do if this fails?

Reply:
May be worth adding the retry logic that QueryGSM has? Would involve adding a delegate for when the response to the creation is received that retries upon failure. Here's QueryGSM example:

```EntityQueryDelegate GSMQueryDelegate;
	GSMQueryDelegate.BindLambda([this, bRetryUntilAcceptingPlayers](const Worker_EntityQueryResponseOp& Op)
	{
		if (Op.status_code != WORKER_STATUS_CODE_SUCCESS)
		{
			UE_LOG(LogGlobalStateManager, Warning, TEXT("Could not find GSM via entity query: %s"), UTF8_TO_TCHAR(Op.message));
		}
		else if (Op.result_count == 0)
		{
			UE_LOG(LogGlobalStateManager, Log, TEXT("GSM entity query shows the GSM does not yet exist in the world."));
		}
		else
		{
			bool bNewAcceptingPlayers = GetAcceptingPlayersFromQueryResponse(Op);

			if (!bNewAcceptingPlayers && bRetryUntilAcceptingPlayers)
			{
				UE_LOG(LogGlobalStateManager, Log, TEXT("Not yet accepting new players. Will retry query for GSM."));
				RetryQueryGSM(bRetryUntilAcceptingPlayers);
			}
			else
			{
				ApplyDeploymentMapDataFromQueryResponse(Op);
			}

			return;
		}

		if (bRetryUntilAcceptingPlayers)
		{
			RetryQueryGSM(bRetryUntilAcceptingPlayers);
		}
	});

	Receiver->AddEntityQueryDelegate(RequestID, GSMQueryDelegate);```

Reply:
This requires creating a new type of delegate and changing `SpatialReceiver` too so I wanted to get thoughts from others if they think it's worth it or the existing error message is enough and we shouldn't try to recover from such errors.

Reply:
I'm not sure what the failure rate for Entity Creation is, maybe we can find out? That being said, it's a fatal error for one of these not to be created, so even if the failure rate is low, we should wrap it in some retry logic to make sure.

Reply:
I see entities failed to be created when there are a large amount of entities being created at the start of the deployment. It could be the case that 100 or so fail or maybe 5. It varies. Would definitely say it needs retry logic.